Sources/SmileID/Classes/DocumentVerification/Model/DocumentCaptureModel.swift

```diff
@@ -1,4 +1,313 @@
+
+        let filteredImage = orientedImage.applyingFilter("CIPerspectiveCorrection", parameters: [
+            "inputTopLeft": CIVector(cgPoint: cartesianScaledQuad.bottomLeft),
+            "inputTopRight": CIVector(cgPoint: cartesianScaledQuad.bottomRight),
+            "inputBottomLeft": CIVector(cgPoint: cartesianScaledQuad.topLeft),
+            "inputBottomRight": CIVector(cgPoint: cartesianScaledQuad.topRight)
+        ])
+        switch side {
+        case .back:
+            backImage = UIImage.from(ciImage: filteredImage)
+        case .front:
+            frontImage = UIImage.from(ciImage: filteredImage)
+        }
+    }
+
+    func resumeCameraSession() {
+        cameraManager.resumeSession()
+    }
+
+    func pauseCameraSession() {
+        cameraManager.pauseSession()
+    }
+
+    func handleCompletion() {
+        switch processingState {
+        case .complete(let response, let error):
+            pauseCameraSession()
+            processingState = .endFlow
+            if let error = error {
+                captureResultDelegate?.didError(error: error)
+                return
+            }
+            if let selfie = selfie {
+                captureResultDelegate?.didSucceed(selfie: selfie,
+                                                  documentFrontImage: frontImage!.jpegData(compressionQuality: 1)!,
+                                                  documentBackImage: backImage!.jpegData(compressionQuality: 1)!,
+                                                  jobStatusResponse: response)
+            }
+        default:
+            break
+        }
+    }
+
+    func submit(navigation _: NavigationViewModel) {
+        if captureBothSides && side == .front {
+            processingState = nil
+            side = .back
+            navigation?.navigate(destination: .documentBackCaptureInstructionScreen(documentCaptureViewModel: self,
+                                                                                   delegate: captureResultDelegate),
+                                style: .push)
+        } else {
+            processingState = .inProgress
+            navigation?.navigate(destination: .selfieCaptureScreen(selfieCaptureViewModel:
+                                                                    SelfieCaptureViewModel(userId: userId,
+                                                                                           jobId: jobId,
+                                                                                           isEnroll: false,
+                                                                                           shoudSubmitJob: false),
+                                                                  delegate: self),
+                                style: .push)
+        }
+    }
+
+    func handleDeclineButtonTap() {
+        processingState = nil
+        
+    }
+
+    func handleRetry() {
+        processingState = .inProgress
+        //TO-DO: Handle retry
```

Comment:
Are you planning on addressing this TODO in this PR? Retry is definitely something we need to have

Reply:
Done